packages/core/admin/admin/src/components/NpsSurvey/index.js

```diff
@@ -123,13 +190,121 @@ const NpsSurvey = () => {
     setSurveyIsShown(false);
   };
 
-  // TODO: replace with the proper UI
   return (
-    <Flex gap={2} padding={2}>
-      <Typography>NPS SURVEY</Typography>
-      <Button onClick={handleDismiss}>Dismiss</Button>
-      <Button onClick={handleSubmitResponse}>Submit</Button>
-    </Flex>
+    <Portal>
+      <Formik
+        initialValues={{ feedback: '', selectedRating: null }}
+        onSubmit={handleSubmitResponse}
+        validationSchema={yup.object({
+          feedback: yup.string(),
+          selectedRating: yup.number(),
+        })}
+      >
+        {({ handleSubmit, values, handleChange, setFieldValue }) => (
+          <Form name="npsSurveyForm" noValidate onSubmit={handleSubmit}>
+            <BannerWrapper hasRadius direction="column" padding={4}>
+              {isFeedbackResponse ? (
+                <Typography fontWeight="semiBold">
+                  {formatMessage({
+                    id: 'app.components.NpsSurvey.feedback-response',
+                    defaultMessage: 'Thank you very much for your feedback!',
+                  })}
+                </Typography>
+              ) : (
+                <>
+                  <Flex justifyContent="space-between" width="100%">
+                    <Header>
+                      <Typography fontWeight="semiBold">
+                        {formatMessage({
+                          id: 'app.components.NpsSurvey.banner-title',
+                          defaultMessage:
+                            'How likely are you to recommend Strapi to a friend or colleague?',
+                        })}
+                      </Typography>
+                    </Header>
+                    <IconButton
+                      onClick={handleDismiss}
+                      aria-label={formatMessage({
+                        id: 'app.components.NpsSurvey.dismiss-survey-label',
+                        defaultMessage: 'Dismiss survey',
+                      })}
+                      icon={<Cross />}
+                    />
+                  </Flex>
+                  <Flex gap={2} marginLeft={8} marginRight={8} marginTop={2} marginBottom={2}>
+                    <Typography variant="pi" textColor="neutral600">
+                      {formatMessage({
+                        id: 'app.components.NpsSurvey.no-recommendation',
+                        defaultMessage: 'Not at all likely',
+                      })}
+                    </Typography>
+                    {ratingArray.map((number) => {
+                      return (
+                        <FieldWrapper
+                          key={number}
+                          className={values.selectedRating === number ? 'selected' : null}
+                        >
+                          <FieldLabel htmlFor={number} id={`${number}-rating`}>
+                            <VisuallyHidden>
+                              <FieldInput
+                                type="radio"
+                                id={number}
```

Comment:
I think plain numbers are not allowed as id attributes. Could we use something namespace here, such as `nps_survey_rating_${number}`?